### PR TITLE
Project 01: Setup Initial Data Model for the Blog App

### PR DESCRIPTION
**Pull request by GhausAliKhan**
### In this branch I did: 

The foundational structure for the "Blog App" data model in alignment with the Microverse Bootcamp guidelines and the provided Entity Relationship Diagram.

- Set up a new Ruby on Rails application with the recommended project structure.
- Created migration files to establish the necessary tables and columns as per the ERD.
- Defined relationships between tables, ensuring foreign key constraints, naming conventions, and indexing.

### New Files & Modifications

- Migration files for `User`, `Post`, `Comment`, and `Like` tables.
- Modified `schema.rb` to reflect the newly created tables and columns.
- Made sure the `User` table contains a `photo` column intended to store links to user images.
- Ensured that the `Post` table references the `User` table with an `author_id` column.

### Conclusion
It prepares the groundwork for the next stages of development, including adding data validations, model specs, and building out controllers and views.

### CODE REVIEW

File: db/migrate/20231122100839_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreatePosts < ActiveRecord::Migration[7.1]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :body
+      t.references :author, null: false, foreign_key: { to_table: :users }
+      t.integer :comments_counter, default: 0
+      t.integer :likes_counter, default: 0
+
```
Comment (mikethreels) [inline db/migrate/20231122100839_create_posts.rb:9]: - [ ] in the post table there should be a column for a photo as well according to the requirements.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Ghaus, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job creating the tables.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20231122100839_create_posts.rb (hunk 1 above)
Comment (GhausAliKhan) [reply db/migrate/20231122100839_create_posts.rb:9]: Hi @mikethreels 
By following the ERD, the column for a photo is not in the posts table its in `users`. 
Can you check again and elaborate more?


image

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @GhausAliKhan,

✔️ Table column types are well defined

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**